File: frontend/src/models/Note.ts
export interface Note {
  id: number
  title: string
  content: string
  current_views: number
  max_views: number
}

export class NoteData {
  title: string
  content: string
  expiration_date?: string
  max_views?: string

  constructor(title: string, content: string, expiration_date?: string, max_views?: string) {
    this.title = title
    this.content = content
    this.expiration_date = expiration_date
    this.max_views = max_views
  }
}

File: backend/models/note.go
package model

import (
	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Note struct {
	gorm.Model
	UniqueUrl      uuid.UUID `json:"unique_url"`
	Title          string    `json:"title"`
	Content        string    `json:"content"`
	ExpirationDate string    `json:"expiration_date"`
	CurrentViews   int       `json:"current_views"`
	MaxViews       int       `json:"max_views"`
	UserID         uint      `json:"user_id"`
}

type CreateNoteInput struct {
	Title          string `json:"title"`
	Content        string `json:"content"`
	ExpirationDate string `json:"expiration_date"`
	MaxViews       int    `json:"max_views"`
}
